lib/rust/profiling/src/lib.rs

```diff
@@ -0,0 +1,398 @@
+//!The performance-logger library allows to create performance reports via the
+//! [Web Performance API](https://developer.mozilla.org/en-US/docs/Web/API/Performance).
+//!
+//! The API provided allows the marking of intervals between a start and end, which the gets
+//! measured and marked in the Chrome DevTools Performance Monitor. Intervals can be assigned a
+//! log level, for example, `TASK`, SECTION` or `DEBUG` to allow filtering based on the current
+//! needs.
+//!
+//! Example usage
+//! --------------
+//! ```
+//! let some_task = || "DoWork";
+//!
+//! // Manually start and end the measurement.
+//! let gui_init = profiling::task::start("GUI initialization");
+//! some_task();
+//! gui_init.end();
+//! // Or use the `measure` method.
+//! profiling::task::measure("GUI initialization", some_task);
+//! ```
+//!
+//! Note that this API and encoding formats for messages are synced with the JS equivalent in
+//! `app/ide-desktop/lib/profiling/src/profiling.ts`.
+#![warn(missing_debug_implementations)]
+#![warn(missing_docs)]
+#![warn(trivial_casts)]
+#![warn(trivial_numeric_casts)]
+#![warn(unsafe_code)]
+#![warn(unused_import_braces)]
+
+use enso_prelude::*;
+
+use enso_prelude::fmt::Formatter;
+use enso_web::performance;
+use inflector::Inflector;
+use ordered_float::OrderedFloat;
+use serde::Deserialize;
+use serde::Serialize;
+use serde_plain::from_str;
+use web_sys::PerformanceEntry;
+
+
+
+// =================
+// === Log Level ===
+// =================
+
+#[derive(Copy, Clone, Debug)]
+#[derive(Serialize, Deserialize)]
+#[allow(missing_docs)]
+pub enum ProfilingLevel {
+    Section,
+    Task,
+    Detail,
+    Debug,
+}
```

Comment:
IMO we should handle it with a proc macro not to repeat ourselves again and keep these names in a single place only (we already have a bug in the codebase that these names are not the same in all places). More info below.

Reply:
I looked into this and I think in its current form there is very little repetition, and it is in a place that can be edited easily: all the changes are next to each other in the file, they fit into a single display and the compiler will even warn you if you missed one of them. 

On the JS side, there is no repetition besides defining the new profiling level. 

To me, it seems that introducing a proc macro and automated code generation on the JS side, would  introduce more complexity than is justified for this.